### PR TITLE
Add requirements.txt

### LINKED ISSUE #299: Add requirements.txt file for developpers
Dependencies for pdfminer are stated in various places:

- package dependencies are in `setup.py`
- test dependencies in `.travis.yml`
- some are never states, like the package Pillow that is used in [on line 87 in image.py](https://github.com/pdfminer/pdfminer.six/blob/41d1efac7ef3ac71164eeb8200af39a2bfa09c59/pdfminer/image.py#L87)

I think it is best to replace those with a single point of truth. We could use a requirements.txt file or a pipenv lock file.

### PR DESCRIPTION
**Pull request by charlesreid1**
Closes #299 

Procedure to create/update requirements.txt:

```
# clone the pdfminer.six repo
git clone https://github.com/pdfminer/pdfminer.six.git
# create a virtual environment
virtualenv vp && source vp/bin/activate`
# install the pdfminer.six package
python setup.py build install
# freeze requirements
pip freeze > requirements.txt
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+chardet==3.0.4
+nose==1.3.7
+pdfminer.six==20181108
```
Comment (pietermarsman) [inline requirements.txt]: pdfminer.six is not a depencency of pdfminer.six ;)

**Comment by pietermarsman** [comment]
@charlesreid1 what do you think of my additional remark:

> Is there any particular reason that you included the versions of those packages? My personal preference is to to specify only the major version of the package.
> 
> Also, like you said, PIL is also a dependency of pdfminerl.six.
> 
> The setup.py should also be updated in this PR.
> 
> The .travis.yml should also be updated by this PR.

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+chardet==3.0.4
+nose==1.3.7
+pycryptodome==3.9.0
+six==1.12.0
```
Comment (hugovk) [inline requirements.txt]: six can be removed after https://github.com/pdfminer/pdfminer.six/pull/305 is merged

**Comment by hugovk** [comment]
Please see https://github.com/pdfminer/pdfminer.six/pull/306 for an alternative.

**Pull request closed without merging**